### ISSUE TITLE
Add retries and confirmations to ensure CNCF runners and machines are removed.

### ISSUE THREAD

**Issue by gyohuangxin** [layer5io/meshery-smp-action#58]
#### Description

There are some remaining CNCF runners not being remove after tests done, the number of them gradually increases over time. 
We can delete them manually, but it's better to make sure they are properly removed.
![image](https://user-images.githubusercontent.com/42127654/179132263-a312392c-33a2-4c23-a1f2-a26f2ab372e1.png)

The same thing happened to equinix servers deletion：
![image](https://user-images.githubusercontent.com/42127654/179133013-d9d568a0-46d9-4081-904f-31ffafb1d6a8.png)

#### Expected Behavior

We should add retries and confirmations to ensure CNCF runners and machines are removed.

#### Environment:
- Meshery Version: 
- Kubernetes Version: 
- Host OS:
- Browser:

**Comment by leecalcote** [comment]
Uh-oh. We do need to complete this item.

**Comment by vielmetti** [comment]
It's possible to create machines on Equinix Metal in such a way that there's a termination time associated with them. See the "termination_time" field at 

https://deploy.equinix.com/developers/api/metal/#tag/Devices/operation/createDevice

in the Equinix Metal API reference. 

(That's not a substitute for cleanup, but it could backstop any other efforts if there's a bug somewhere else).

**Comment by vielmetti** [comment]
There was a short-lived API outage yesterday, described at

https://status.equinixmetal.com/incidents/h30n2jlr5d3p

which may have impacted manual deletion of these systems. Please retry if you were affected by this. As of this writing, there are 48 systems deployed.

**Comment by gyohuangxin** [comment]
@vielmetti I'm still facing the issue to access the management UI: 
![image](https://user-images.githubusercontent.com/42127654/222205783-a5145050-4bc6-40ef-a468-2c0b25f398d4.png)

**Comment by vielmetti** [comment]
@gyohuangxin can you open up a ticket with our support team? I'll share your UI issue with the team, but it may be something specific to your account.

**Comment by vielmetti** [comment]
@gyohuangxin Can you please task someone else on the project to assist you with cleaning up the idle and stranded resources while we sort out your access problems.

**Comment by leecalcote** [comment]
All existing servers were manually deprovisioned today. A fresh batch of newly provisioned servers is running (now) from workflow schedule. Let's see if those servers are automatically deprovisioned on completion of their task.

**Comment by leecalcote** [comment]
Yes, it seems that the test servers are successfully deprovisioned at end of test. 👍
